oak/server/storage/spanner_provider.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OAK_SERVER_STORAGE_SPANNER_PROVIDER_H_
+#define OAK_SERVER_STORAGE_SPANNER_PROVIDER_H_
+
+#include "google/spanner/v1/spanner.grpc.pb.h"
+#include "storage_provider.h"
+
+namespace oak {
+
+// SpannerProvider is an implementation of StorageProvider using Cloud Spanner.
+class SpannerProvider final : public StorageProvider {
+ public:
+  explicit SpannerProvider(std::shared_ptr<grpc::ChannelInterface> channel);
+
+  grpc::Status Read(const ReadRequest* request, ReadResponse* response) override;
```

Comment:
Do these actually need to appear in this .h file? I thought it would be fine to just override them in the .cc file, since they are coming from the superclass. Anyway, good to have them for extra clarity I suppose!

Reply:
Yes, they do need to be declared in the derived class as well.